src/components/data/HangingGarden/hooks/useHangingGardenData.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import moment from 'moment';
+import { HttpResponse } from '@equinor/fusion/lib/http/HttpClient';
+
+const useHangingGardenData = <T, C extends keyof ApiClients, E extends keyof ApiClients[C]>(
+    client: C,
+    endpoint: E,
+    applyToFetchedData?: ((data: T[]) => T[]) | null,
+    searchableValues?: ((data: T) => T) | null
+) => {
+    const currentContext = useCurrentContext();
+    const api = useApiClients();
+    const [data, setData] = React.useState<T[]>([]);
+    const [error, setError] = React.useState<Error | null>(null);
+    const [isFetching, setIsFetching] = React.useState(false);
+    const [cacheAgeDate, setCacheAgeDate] = React.useState<Date>(new Date());
+    const [cacheDuration, setCacheDuration] = React.useState<number>(30);
+    const [cacheAge, setCacheAge] = React.useState('');
+
+    const [cacheIsInvalid, setCacheIsInvalid] = React.useState(true);
+    const cacheCheckTimerRef = React.useRef<NodeJS.Timeout>();
+
+    const checkCacheValidity = React.useCallback(() => {
+        setCacheAge(moment(cacheAgeDate).fromNow());
+        const isInvalid =
+            new Date(cacheAgeDate).setMinutes(cacheAgeDate.getMinutes() + cacheDuration) <
+            Date.now();
+        setCacheIsInvalid(isInvalid);
+        cacheCheckTimerRef.current = setTimeout(checkCacheValidity, 60000);
+    }, [cacheAgeDate, cacheDuration]);
+
+    React.useEffect(() => {
+        if (cacheCheckTimerRef.current) {
+            clearTimeout(cacheCheckTimerRef.current);
+        }
+
+        checkCacheValidity();
+
+        return () => {
+            if (cacheCheckTimerRef.current) {
+                clearTimeout(cacheCheckTimerRef.current);
+            }
+        };
+    }, [checkCacheValidity]);
+
+    const reset = React.useCallback(() => {
+        setData([]);
+        setError(null);
+        setIsFetching(false);
+    }, []);
+
+    React.useEffect(() => {
+        reset();
+    }, [currentContext]);
+
+    const getData = React.useCallback(
+        async (currentContext: Context, invalidateCache: boolean) => {
+            const response = (await (api[client][endpoint] as any)(
+                currentContext.id,
+                invalidateCache
+            )) as HttpResponse<T[]>;
+
+            const cacheAgeHeader = response.headers.get('x-pp-cache-age');
+            const cacheAge =
+                cacheAgeHeader === null || cacheAgeHeader === 'live'
+                    ? new Date()
+                    : new Date(cacheAgeHeader);
+            const cacheDurationHeader = response.headers.get('x-pp-cache-duration-minutes');
+            const cacheDuration = parseInt(cacheDurationHeader || '30', 10);
+
+            return {
+                data: response.data,
+                cacheAge,
+                cacheDurationInMinutes: cacheDuration,
+            };
+        },
+        [client, endpoint, api, currentContext]
+    );
```

Comment:
create a hook for this, same method is used other places, think procosys.ts

Reply:
I dont understand this comment. 
How is the getData used somewhere else ? 
How does it relate to procosys.ts ?  
Thats just used by the story. What statuses and colors it should use etc.

Reply:
new hook useHagingGardenGetData in lack of better names. 
Fetched data from api and delivers back the data and cacheDuration info. 

Outside of cleaning up the cachedurtion, there is no other data manipulation going on here. 
That is stil done in the useHagingGardenData, which now implements the above hook